### PR TITLE
Fix: Wrap routeableipv4 in try catch

### PR DESCRIPTION
**Pull request by covert8**
#### Description
Wrap NetworkInterface.GetAllNetworkInterfaces in try catch to solve race condition on aarch64

#### Database Migration
NO

#### Issues Fixed or Closed by this PR
* Closes #
https://github.com/Prowlarr/Prowlarr/issues/2076

### CODE REVIEW

**Review by markus101 — APPROVED** [review]
Should we log anything so people know what's going on without having to dig into code?

**Comment by covert8** [comment]
Sounds like a good idea

File: src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs [hunk 1]
```diff
@@ -285,7 +300,9 @@ private static async ValueTask<Stream> onConnect(SocketsHttpConnectionContext co
                 catch
                 {
                     // Do not retry IPv6 if a routable IPv4 address is available, otherwise continue to attempt IPv6 connections.
-                    useIPv6 = !HasRoutableIPv4Address();
+                    var routableIPv4 = HasRoutableIPv4Address();
+                    _logger.Info("Fallback to ipv4, disabling ipv6: {routableIPv4}");
```
Comment (markus101) [inline src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs:304]:
```suggestion
                    _logger.Info("IPv4 is available: {0}, IPv6 will be {1}", routableIPv4, routableIPv4 ? "disabled", "left enabled");
```

File: src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs [hunk 2]
```diff
@@ -252,13 +258,22 @@ private CredentialCache GetCredentialCache()
...
-                    !IPAddress.IsLoopback(ip.Address)));
+                return networkInterfaces.Any(ni =>
+                    ni.OperationalStatus == OperationalStatus.Up &&
+                    ni.GetIPProperties().UnicastAddresses.Any(ip =>
+                        ip.Address.AddressFamily == AddressFamily.InterNetwork &&
+                        !IPAddress.IsLoopback(ip.Address)));
+            }
+            catch (Exception e)
+            {
+                _logger.Debug("Caught exception while GetAllNetworkInterfaces assuming IPv4 connectivity");
```
Comment (markus101) [inline src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs:273]: Be worth logging the exception too.

File: src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs [hunk 3]
```diff
@@ -252,13 +258,22 @@ private CredentialCache GetCredentialCache()
...
+                _logger.Trace("HasRoutableIPv4Address: " + e.Message);
```
Comment (markus101) [inline src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs:274]: I don't think we need to log this separately, could append to the debug message.

File: src/NzbDrone.Core.Test/Framework/CoreTest.cs [hunk 4]
```diff
@@ -25,7 +25,7 @@ protected void UseRealHttp()
             Mocker.SetConstant<IHttpProxySettingsProvider>(new HttpProxySettingsProvider(Mocker.Resolve<ConfigService>()));
             Mocker.SetConstant<ICreateManagedWebProxy>(new ManagedWebProxyFactory(Mocker.Resolve<CacheManager>()));
             Mocker.SetConstant<ICertificateValidationService>(new X509CertificateValidationService(Mocker.Resolve<ConfigService>(), TestLogger));
-            Mocker.SetConstant<IHttpDispatcher>(new ManagedHttpDispatcher(Mocker.Resolve<IHttpProxySettingsProvider>(), Mocker.Resolve<ICreateManagedWebProxy>(), Mocker.Resolve<ICertificateValidationService>(), Mocker.Resolve<UserAgentBuilder>(), Mocker.Resolve<CacheManager>()));
+            Mocker.SetConstant<IHttpDispatcher>(new ManagedHttpDispatcher(Mocker.Resolve<IHttpProxySettingsProvider>(), Mocker.Resolve<ICreateManagedWebProxy>(), Mocker.Resolve<ICertificateValidationService>(), Mocker.Resolve<UserAgentBuilder>(), Mocker.Resolve<CacheManager>(),Mocker.Resolve<IHttpDispatcher>(), TestLogger));
```
Comment (markus101) [inline src/NzbDrone.Core.Test/Framework/CoreTest.cs:28]:
```suggestion
            Mocker.SetConstant<IHttpDispatcher>(new ManagedHttpDispatcher(Mocker.Resolve<IHttpProxySettingsProvider>(), Mocker.Resolve<ICreateManagedWebProxy>(), Mocker.Resolve<ICertificateValidationService>(), Mocker.Resolve<UserAgentBuilder>(), Mocker.Resolve<CacheManager>(), Mocker.Resolve<IHttpDispatcher>(), TestLogger));
```

**Review by markus101 — CHANGES_REQUESTED** [review]
Converting these methods to non-static seems reasonable, shouldn't make too much of an impact over all.

**Review by markus101 — APPROVED** [review]

**Pull request merged**